mongoengine/fields.py

```diff
@@ -675,6 +676,16 @@ def validate(self, value, clean=True):
             value.validate(clean=clean)
 
 
+class OrderedDynamicField(DynamicField):
+    """A field that wraps DynamicField. This uses OrderedDict class
+    to guarantee to store data in the defined order instead of dict.
+    """
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Instead of adding another field class, would it make sense add an optional argument to the original `DynamicField`'s constructor? For instance, `field_name = DynamicField(ordered=True)`, or - to provide even more extensibility - `field_name = DynamicField(container_class=OrderedDict)`.